### PR TITLE
USB: webusb: Add MS-OS descriptors for compatible IDs

### PR DESCRIPTION
**Pull request by finikorg**
Amended PR #7029. 
Added the last patch on top of original PR.

@IvanSanchez

### CODE REVIEW

**Comment by carlescufi** [comment]
CC @pawelzadrozniak @rakons

File: samples/subsys/usb/webusb/src/main.c [hunk 1]
```diff
@@ -18,37 +18,101 @@
...
+			0xC7, 0x4C,
+			0x9C, 0xD2,
+			0x65, 0x9D, 0x9E, 0x64, 0x8A, 0x9F,
+		},
+	},
+	.capability_data_msos = {
+		/* Windows version (8.1) (0x06030000) */
+		.dwWindowsVersion = sys_cpu_to_le32(0x06030000),
+		.wMSOSDescriptorSetTotalLength =
+			sys_cpu_to_le16(0x0A + 0x14 + 0x08),
```
Comment (jfischer-phytec-iot) [inline samples/subsys/usb/webusb/src/main.c]: What is 0x0A bytes long, can you use sizeof here?

Comment (finikorg) [reply samples/subsys/usb/webusb/src/main.c]: fixed

**Review by carlescufi — APPROVED** [review]
From my understanding of @IvanSanchez's work this looks fine to me.

**Comment by carlescufi** [comment]
@jfischer-phytec-iot can you re-review this?

**Comment by jfischer-phytec-iot** [comment]
@carlescufi just made, LGTM, but I guess the last 2 commits can be merged into fb742e6

**Comment by carlescufi** [comment]
@jfischer-phytec-iot I think the reason not to is that they were originally by another author, @IvanSanchez

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**